torch/csrc/dynamo/guards.cpp

```diff
@@ -895,6 +895,41 @@ static PyObject* assert_size_stride(PyObject* dummy, PyObject* args) {
   Py_RETURN_TRUE;
 }
 
+static PyObject* assert_alignment(PyObject* dummy, PyObject* args) {
```

Comment:
since we already have assert_sizes_strides, is it worth extending that ?

Reply:
I think it's more flexible to keep them separate.

size/strides asserts applies to all tensors while alignment asserts is only applied to tensors that the Inductor think is aligned at compile time.  If Inductor already assumes a tensor is not-aligned, there is no assertion needed.